helmclient.go

```diff
@@ -815,23 +816,31 @@ func getTillerImage(pod *corev1.Pod) (string, error) {
 	return tillerImage, nil
 }
 
-func isTillerOutdated(pod *corev1.Pod) (bool, error) {
-	currentTillerImage, err := getTillerImage(pod)
+func validateTillerVersion(pod *corev1.Pod, desiredImage string) error {
+	currentImage, err := getPodImage(pod)
 	if err != nil {
-		return false, microerror.Mask(err)
+		return microerror.Mask(err)
 	}
 
-	currentTillerVersion, err := parseTillerVersion(currentTillerImage)
+	currentVersion, err := parseTillerVersion(currentImage)
 	if err != nil {
-		return false, microerror.Mask(err)
+		return microerror.Mask(err)
 	}
 
```

Comment:
I noticed that below we were using `TillerImageSpec` const which we shouldn't as we specify `tillerImage` field now. This probably breaks some tests.

Reply:
Unit tests fixed.